src/components/application_manager/test/commands/mobile/set_global_properties_request_test.cc

```diff
@@ -366,6 +404,39 @@ TEST_F(SetGlobalPropertiesRequestTest, Run_NoVR_SUCCESS) {
   command->Run();
 }
 
+TEST_F(SetGlobalPropertiesRequestTest, Run_VRCouldNotGenerate_INVALID_DATA) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject keyboard_properties(smart_objects::SmartType_Map);
+  (*msg)[strings::msg_params][hmi_request::keyboard_properties] =
+      keyboard_properties;
+  SmartObject menu_title("Menu_Title");
+  (*msg)[strings::msg_params][hmi_request::menu_title] = menu_title;
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_message_helper_, VerifyImageVrHelpItems(_, _, _)).Times(0);
+  EXPECT_CALL(mock_app_manager_,
+              RemoveAppFromTTSGlobalPropertiesList(kConnectionKey));
+  SmartObject* vr_help_title = NULL;
+  CommandsMap commands_map;
+  SmartObject empty_msg(smart_objects::SmartType_Map);
+  commands_map.insert(std::pair<uint32_t, SmartObject*>(1u, &empty_msg));
+  DataAccessor<CommandsMap> accessor(commands_map, lock_);
+  EXPECT_CALL(*mock_app_, commands_map()).WillOnce(Return(accessor));
+  EXPECT_CALL(*mock_app_, vr_help_title()).WillOnce(Return(vr_help_title));
+  EXPECT_CALL(*mock_app_, set_menu_title(_)).Times(0);
+
+  SharedPtr<SetGlobalPropertiesRequest> command(
+      CreateCommand<SetGlobalPropertiesRequest>(msg));
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::INVALID_DATA),
```

Comment:
@LuxoftAKutsan , because there is no matcher for concrete result_code.  `MobileResultCodeIs` takes as arg expected result_code and checks it with value in first arg in `ManageMobileCommand`.
Do not think that we need matchers for every result_code